### PR TITLE
[MacOS] Add software to MacOS 12 image was not available during the Monterey beta stage.

### PR DESCRIPTION
**Pull request by shilovmaksim**
# Description
Some software was not available during the Monterey beta stage. Adding software already available.
- [x] Carthage
- [x] Clang/LLVM
- [ ] Conda
- [x] GnuPG
- [ ] Helm
- [x] PHP
- [x] Nomad shenzhen CLI
- [x] SoX

## Check list
- [x] Related issue / work item is attached
- [x] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [x] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/macos/software-report/SoftwareReport.Generator.ps1 [hunk 1]
```diff
@@ -43,21 +43,17 @@ $languageAndRuntimeList = @(
     (Get-DotnetVersionList),
     (Get-GoVersion),
     (Get-JuliaVersion),
-    (Get-KotlinVersion)
+    (Get-KotlinVersion),
+    (Get-PHPVersion),
+    (Get-FortranVersion),
+    (Get-ClangLLVMVersion)
+    (Get-PHPVersion)
```
Comment (al-cheb) [inline images/macos/software-report/SoftwareReport.Generator.ps1:50]: Duplicate 47&50?

Comment (shilovmaksim) [reply images/macos/software-report/SoftwareReport.Generator.ps1:50]: Oh. Indeed. Will fix.

**Review by dmitry-shibanov — DISMISSED** [review]

File: images/macos/software-report/SoftwareReport.Generator.ps1 [hunk 2]
```diff
@@ -139,16 +130,13 @@ $utilitiesList = @(
...
-if ($os.IsLessThanMonterey) {
-    $utilitiesList += @(
-        (Get-GPGVersion),
-        (Get-HelmVersion)
-    )
-}
-
 if ($os.IsHigherThanMojave -and $os.IsLessThanMonterey) {
     $utilitiesList += @(
         (Get-NewmanVersion)
```
Comment (al-cheb) [inline images/macos/software-report/SoftwareReport.Generator.ps1:142]: Out of curiosity, is newman still unsupported or deprecated?

Comment (shilovmaksim) [reply images/macos/software-report/SoftwareReport.Generator.ps1:142]: I'm not really sure. I just walked through the list in the workitem and the list of canary tests I need to update =)

File: images/macos/software-report/SoftwareReport.Generator.ps1 (hunk 1 above)
Comment (shilovmaksim) [reply images/macos/software-report/SoftwareReport.Generator.ps1]: Done

**Review by al-cheb — APPROVED** [review]

**Review by mikhailkoliada — APPROVED** [review]

**Review by dmitry-shibanov — DISMISSED** [review]

**Comment by shilovmaksim** [comment]
Removed "helm" and "miniconda" since they are in the list of software to deprecate: https://github.com/actions/virtual-environments-internal/issues/2647

**Review by MaksimZhukov — APPROVED** [review]

**Pull request merged**